packages/build-utils/src/workspaces/get-workspaces.ts

```diff
@@ -0,0 +1,65 @@
+import { DetectorFilesystem } from '../detectors/filesystem';
+import { workspaceManagers } from './workspace-managers';
+import { detectFramework as detectWorkspaceManagers } from '../detect-framework';
+
+const MAX_DEPTH_TRAVERSE = 2;
```

Comment:
How did you decide on 2?

Reply:
This is based on the most common folder structures for monorepos 

https://www.notion.so/Monorepo-Onboarding-826222fe6849424fa9daab21d3ae6468?d=e41743eb2fde4c628c2a15c00e415b57#20c28966c729422f9c2ebf2208cf0f6b

Reply:
It looks like a depth of `3` was still being discussed. Should this be `2` or `3`?

Reply:
For workspace detection, it will remain at 2.

For non-workspace monorepos, it will be 3.  It's covered in this PR https://github.com/vercel/vercel/pull/7826